crates/host/src/wasmbus/mod.rs

```diff
@@ -425,6 +445,38 @@ impl Lattice {
             .build()
             .context("failed to build runtime")?;
         let event_builder = EventBuilderV10::new().source(host_key.public_key());
+
+        let jetstream = async_nats::jetstream::new(nats.clone());
+        // TODO: Use prefix
+        let bucket = format!("LATTICEDATA_{prefix}", prefix = "default");
+        jetstream
+            .create_stream(async_nats::jetstream::stream::Config {
```

Comment:
This isn't quite right:
1) We shouldn't be creating a stream manually, we should be creating a bucket
2) We should only attempt to create the bucket if it doesn't already exist. E.g. the first host started in a lattice can create the bucket; subsequent hosts should re-use the existing bucket
3) When we construct the JS client we need to pass the JS domain

Reply:
1. This is exactly what OTP does, if we do not do it exactly the same way - OTP and Rust host cannot coexist in the same lattice
2. That's exactly how it works - it does not do anything if the stream already exists
3. That will come later together with all the other config (like lattice prefixes)

Reply:
For context, please read the implementation of this method https://docs.rs/async-nats/latest/async_nats/jetstream/context/struct.Context.html#method.create_key_value

The method itself does not provide means to specify all the options that OTP specifies (which is a requirement for this effort) - we have to use a lower-level one

Reply:
1) The code you referenced for OTP should have the same effect as creating a bucket. See the equivalent from the NATS CLI commands:

```
nats stream add KV_foobar
? Subjects $KV.foobar.>
? Storage file
? Replication 1
? Retention Policy Limits
? Discard Policy New
? Stream Messages Limit -1
? Per Subject Messages Limit 1
? Total Stream Size -1
? Message TTL -1
? Max Message Size -1
? Duplicate tracking time window 2m0s
? Allow message Roll-ups Yes
? Allow message deletion No
? Allow purging subjects or the entire stream Yes
Stream KV_foobar was created

Information for Stream KV_foobar created 2023-07-26 12:34:43

             Subjects: $KV.foobar.>
             Replicas: 1
              Storage: File

Options:

            Retention: Limits
     Acknowledgements: true
       Discard Policy: New
     Duplicate Window: 2m0s
    Allows Msg Delete: false
         Allows Purge: true
       Allows Rollups: true

Limits:

     Maximum Messages: unlimited
  Maximum Per Subject: 1
        Maximum Bytes: unlimited
          Maximum Age: unlimited
 Maximum Message Size: unlimited
    Maximum Consumers: unlimited


State:

             Messages: 0
                Bytes: 0 B
             FirstSeq: 0
              LastSeq: 0
     Active Consumers: 0
```

```
nats stream ls -a
╭────────────────────────────────────────────────────────────────────────────────╮
│                                    Streams                                     │
├───────────┬─────────────┬─────────────────────┬──────────┬──────┬──────────────┤
│ Name      │ Description │ Created             │ Messages │ Size │ Last Message │
├───────────┼─────────────┼─────────────────────┼──────────┼──────┼──────────────┤
│ KV_foobar │             │ 2023-07-26 12:24:23 │ 0        │ 0 B  │ never        │
╰───────────┴─────────────┴─────────────────────┴──────────┴──────┴──────────────╯
```

```
nats kv add foobar
nats: error: nats: stream name already in use
```

```
nats kv del foobar
? Deleted bucket foobar? Yes

nats kv add foobar
Information for Key-Value Store Bucket foobar created 2023-07-26T12:31:35-06:00

Configuration:

          Bucket Name: foobar
         History Kept: 1
        Values Stored: 0
   Backing Store Kind: JetStream
          Bucket Size: 0 B
  Maximum Bucket Size: unlimited
   Maximum Value Size: unlimited
     JetStream Stream: KV_foobar
              Storage: File

  Cluster Information:

                Name:
              Leader: NA3ASHNOFP6SXS2ZVBTBLAXGFF2YJS3FFS2VZFZ7TZR6B5LENER3RK7H

nats stream info KV_foobar
Information for Stream KV_foobar created 2023-07-26 12:35:18

             Subjects: $KV.foobar.>
             Replicas: 1
              Storage: File

Options:

            Retention: Limits
     Acknowledgements: true
       Discard Policy: New
     Duplicate Window: 2m0s
           Direct Get: true
    Allows Msg Delete: false
         Allows Purge: true
       Allows Rollups: true

Limits:

     Maximum Messages: unlimited
  Maximum Per Subject: 1
        Maximum Bytes: unlimited
          Maximum Age: unlimited
 Maximum Message Size: unlimited
    Maximum Consumers: unlimited


State:

             Messages: 0
                Bytes: 0 B
             FirstSeq: 0
              LastSeq: 0
     Active Consumers: 0
```

The only difference between the two is `Direct Get: true`, which we don't care about.

KV buckets _are_ streams under the hood. The OTP code was creating a stream IIRC because the jetstream client we used in Elixir didn't have support for an option we wanted; something like that. I can confirm creating a bucket works as intended because there are wasmCloud lattices I've pre-provisioned using the bucket setup, both with the CLI and with async-nats.

2) If you try to create a stream that already exists, the server will return an error:

```
nats stream add KV_foobar
? Subjects $KV.foobar.>
? Storage file
? Replication 1
? Retention Policy Limits
? Discard Policy New
? Stream Messages Limit -1
? Per Subject Messages Limit -1
? Total Stream Size -1
? Message TTL -1
? Max Message Size -1
? Duplicate tracking time window 2m0s
? Allow message Roll-ups No
? Allow message deletion Yes
? Allow purging subjects or the entire stream Yes
nats: error: could not create Stream: stream name already in use with a different configuration (10058)
```

Unless you use the exact same configuration as what already exists, in which case it is a no-op. To be safe, we should check for the existence of the bucket first before trying to create it.

3) This can wait, yes. Are you tracking TODOs somewhere?